Orange/data/table.py

```diff
@@ -112,26 +112,33 @@ def __setitem__(self, key, value):
                 self.table.metas[self.row_index, -1 - key] = value
 
     def _str(self, limit):
-        def sp_values(matrix, variables):
+        def sp_values(matrix, variables, show_labels=True):
             if not sp.issparse(matrix):
                 return Instance.str_values(matrix[row], variables, limit)
-            begptr, endptr = matrix.indptr[row:row + 2]
-            rendptr = endptr if not limit else min(endptr, begptr + 5)
-            variables = [variables[var]
-                         for var in matrix.indices[begptr:rendptr]]
+
+            # Sparse matrices can't handle slices where an index is out of
+            # bounds like numpy arrays can, so we must determine largest index
+            _, columns = matrix.shape
+            max_idx, has_more_columns = min(5, columns), columns > 5
+
             s = ", ".join(
-                "{}={}".format(var.name, var.str_val(val))
-                for var, val in zip(variables, matrix.data[begptr:rendptr]))
-            if limit and rendptr != endptr:
+                ("%s=" % var.name if (var.is_discrete or matrix[row, idx]) and
```

Comment:
Not sure how I only spotted this now, but this only conditions the variable name on the value `matrix[row, idx]` being non-zero; while the value itself is still printed. Check this:
<img width="1330" alt="screen shot 2017-07-11 at 14 15 05" src="https://user-images.githubusercontent.com/713026/28067995-f8bdad54-6643-11e7-99af-536ed840d6f2.png">